### PR TITLE
Expose @component decorator

### PR DESCRIPTION
**Pull request by bstrie**
While playing around with esper I noticed that making components involved a bit of boilerplate:
```python
class Position:
    def __init__(self, x=0.0, y=0.0, z=0.0):
        self.x = x
        self.y = y
        self.z = z
```
I solved this by using Python's `dataclasses.dataclass` decorator, imported as `component`:
```python
@component
class Position:
    x: float = 0.0
    y: float = 0.0
    z: float = 0.0
```
I thought this looked a bit nicer, so here's a patch that exposes this as `esper.component`, adds a note in the readme, and updates some examples to use it.

### CODE REVIEW

**Comment by benmoran56** [comment]
Thanks for the pull request! 
I like this idea.  The only concern I have is that Python 3.6 is still in support until the end of the year.
There _is_ a backported `dataclasses` module available on PyPi however, so I still think it's OK to move forward with this. 
I'll just put a note in the README for the next release.

File: examples/benchmark_cache.py [hunk 1]
```diff
@@ -4,13 +4,8 @@
 import sys
 import time
 import optparse
-import esper
+from esper import component, Processor, World
 
-try:
```
Comment (benmoran56) [inline examples/benchmark_cache.py:9]: Any reason for the removable of matplotlib here?

**Review by benmoran56 — COMMENTED** [review]
Just wondering why matplotlib was removed.

File: examples/benchmark_cache.py (hunk 1 above)
Comment (bstrie) [reply examples/benchmark_cache.py:9]: Oh I just took that out to test on this machine which doesn't have matplotlib. 🤣 I'll undo that and rebase.

**Comment by bstrie** [comment]
Rebased.

**Review by benmoran56 — APPROVED** [review]

**Comment by benmoran56** [comment]
Thank you very much! I'll try to get a new release out soon.

**Pull request merged**